### ISSUE TITLE
Type 'any' is not assignable to type 'never' (Using Typescript)

### ISSUE THREAD

**Issue by cairoramos7** [expo/vector-icons#137]
When I use the tag **`<Icon name =""/>`** I am always getting an alert from lint saying that type 'any' is not assignable to type 'never' (Image attached).

Would this be a typing problem or is it necessary to do some additional configuration?

System:
Typescript: **`3.9.6`**
Expo SDK: **`38.0.8`**

Alert:
```
Type 'string' is not assignable to type 'never'.ts(2322)
createIconSet.d.ts(37, 5): The expected type comes from property 'name' which is declared here on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<IconProps<never>, {}, any>> & Pick<Readonly<IconProps<...>> & Readonly<...>, "style" | ... 37 more ... | "textBreakStrategy"> & Pick<...> & Pick<...>'
```

![Screen Shot 2020-07-21 at 09 56 59](https://user-images.githubusercontent.com/15651234/88057670-c875ec80-cb38-11ea-86d4-3c88f30dd5aa.png)

**Comment by piyuaryan** [comment]
yes, this is due to the type definition issue,
`createIconSetFromIcoMoon.d.ts` has following defintion

```
export default function (config: any, expoFontName: any, expoAssetId: any): import("./createIconSet").Icon<never, any>;
```

Return type is `<never, any>` , It should be <string, any> or to be vey generic <any,  any>

```
export default function (config: any, expoFontName: any, expoAssetId: any): import("./createIconSet").Icon<string any>;
```

@ide , is it possible to add this fix asap? I've also upgraded the Expo SDK and getting TS error and hence my build pipelines are failing.
Look forward for a fix.
Thanks

**Comment by piyuaryan** [comment]
I tried to create a branch and create PR but looks like I don't have permissions

So attaching changes here:
```
M       build/createIconSetFromIcoMoon.d.ts
M       build/createIconSetFromIcoMoon.js.map
M       src/createIconSetFromIcoMoon.ts
```

[createIconSetFromIcoMoon-Fix.zip](https://github.com/expo/vector-icons/files/5115327/createIconSetFromIcoMoon-Fix.zip)

**Comment by ide** [comment]
You don't need permissions to make a PR, AFAIK all public GH repos let anyone submit a PR.

**Comment by piyuaryan** [comment]
Yeah, I thought so, I tried to create a branch and push, Git gave me error I do not have permission to do so.
That's why I had to post comment here.

Let me try that again @ide 
Thanks for prompt response.
Cheers

**Comment by piyuaryan** [comment]
Done, PR created, Somehow my public keys were deleted from my github account, hence it was giving me permission denied error.
All good now.
PR created.

https://github.com/expo/vector-icons/pull/140

**Comment by brentvatne** [comment]
I released this in @expo/vector-icons@10.2.1 - thanks!
